### PR TITLE
fix: `stackit beta server` commands - add nil pointer checks and tests for the outputResult functions

### PR DESCRIPTION
**Pull request by marceljk**
- add nil pointer checks
- add tests for the outputResult functions within the server commands

### CODE REVIEW

File: internal/cmd/beta/server/backup/create/create_test.go [hunk 1]
```diff
@@ -208,3 +208,31 @@ func TestBuildRequest(t *testing.T) {
...
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name:    "empty",
+			args:    args{},
+			wantErr: false,
+		},
```
Comment (Fyusel) [inline internal/cmd/beta/server/backup/create/create_test.go:227]: Should we add more test cases here? Like for each output format? (none, json, yaml)

Comment (Fyusel) [reply internal/cmd/beta/server/backup/create/create_test.go:227]: Also for the other tests

File: internal/cmd/beta/server/os-update/create/create.go [hunk 2]
```diff
@@ -60,8 +62,19 @@ func NewCmd(p *print.Printer) *cobra.Command {
 				return err
 			}
 
+			serverLabel := model.ServerId
+			// Get server name
+			if iaasApiClient, err := iaasClient.ConfigureClient(p); err == nil {
+				serverName, err := iaasUtils.GetServerName(ctx, iaasApiClient, model.ProjectId, model.ServerId)
+				if err != nil {
+					p.Debug(print.ErrorLevel, "get server name: %v", err)
+				} else if serverName != "" {
```
Comment (Fyusel) [inline internal/cmd/beta/server/os-update/create/create.go:71]: nitpick: different code style, sometimes "else if" sometimes "only if"

File: internal/cmd/beta/server/backup/create/create_test.go (hunk 1 above)
Comment (marceljk) [reply internal/cmd/beta/server/backup/create/create_test.go:227]: In the beginning I also added tests for json and yaml, but in general they can handle everything without panic or returning an error. So even if a pointer is nil and the nil check is removed, they return null as a result, which is fine from myside. And the part for json and yaml output is always the same. Only the default output is always different with our custom code, which needs to be tested to make sure there is no nil check missing.
But we can discuss this in our daily.

File: internal/cmd/beta/server/os-update/create/create.go (hunk 2 above)
Comment (marceljk) [reply internal/cmd/beta/server/os-update/create/create.go:71]: I take a second look and make this more consistent

Comment (marceljk) [reply internal/cmd/beta/server/os-update/create/create.go:71]: I updated it.
Please keep aware that there are still two different styles to get the server name.
This is because in some cases the iaasClient is needed anyway, and the command will fail if it can't configure it. And in some cases the iaasClient is only needed to get the server name. If it fails in the second case, then it can be ignored and the server-id will be taken. So the handling is a bit different there

**Review by Fyusel — APPROVED** [review]
